pkg/block/s3/adapter.go

```diff
@@ -109,6 +116,29 @@ func NewAdapter(s3 s3iface.S3API, opts ...func(a *Adapter)) *Adapter {
 	return a
 }
 
+func (a *Adapter) getBucketRegion(ctx context.Context, bucket string) string {
```

Comment:
This isn't thread safe, is it possible to reach here from different concurrent paths on startup? Also, the locking will also save you from the need to call s3 for the same request over and over again.